posthog/api/team.py

```diff
@@ -673,6 +690,55 @@ def _update_marketing_analytics_config(self, instance: Team, validated_data: dic
         self._capture_diff(instance, "marketing_analytics_config", old_config, new_config)
         return instance
 
+    def _verify_update_session_recording_retention_period(self, instance: Team, new_retention_period: str):
+        retention_feature = instance.organization.get_available_feature(AvailableFeature.SESSION_REPLAY_DATA_RETENTION)
+
+        if not retention_feature:
+            raise ImproperlyConfigured("No retention period feature found.")
```

Comment:
i've got a feeling we've got a "you need to pay us" exception we could use here

Reply:
ah, no, in some places we set `status=status.HTTP_402_PAYMENT_REQUIRED`
no strong opinion

Reply:
All types of users should have a retention feature really - so this is more of a configuration issue than a permissions issue I think.

Reply:
Changed it to an HTTP 500 but with a descriptive message - if the retention feature fails to parse into something useful then that's an internal failure of our system that should not occur and cannot be fixed by the client.